Functions/Admin/Register-RsPowerBI.ps1

```diff
@@ -0,0 +1,178 @@
+            Use the "Connect-RsReportServer" function to set/update a default value.
+        
+        .EXAMPLE
+            PS C:\> Register-RsPowerBI -ClientId 'value1' -ClientSecret 'value2' -AppObjectId 'value3' -TenantName 'value4' -TenantId 'value5' -RedirectUrls 'value6'
+    
+            <Insert description here>
+        
+        .NOTES
+            Author:      ???
+            Editors:     Friedrich Weinmann
+            Created on:  ???
+            Last Change: 26.01.2017
+            Version:     1.1
+            
+            Release 1.1 (26.01.2017, Friedrich Weinmann)
+            - Fixed Parameter help (Don't poison the name with "(optional)", breaks Get-Help)
+            - Implemented ShouldProcess (-WhatIf, -Confirm)
+            - Replaced calling exit with throwing a terminating error (exit is a bit of an overkill when failing a simple execution)
+            - Improved error message on failure.
+            - Standardized the parameters governing the Report Server connection for consistent user experience.
+            - Added input validation, to ensure no out-of-date servers are specified.
+            - Renamed function from "Register-PowerBI" to "Register-RsPowerBI", in order to conform to naming standards and include the module prefix. Introduced an alias with the old name for backwards compatibility.
+            
+            Release 1.0 (???, ???)
+            - Initial Release
+        
+        .LINK
+            https://blogs.msdn.microsoft.com/sqlrsteamblog/2016/07/22/manually-configuring-power-bi-integration-in-reporting-services/
+    #>
+
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+    param
+    (
+        [Parameter(Mandatory = $True)]
+        [string]
+        $ClientId,
+
+        [Parameter(Mandatory = $True)]
+        [string]
+        $ClientSecret,
+        
+        [Parameter(Mandatory = $True)]
+        [string]
+        $AppObjectId,
+        
+        [Parameter(Mandatory = $True)]
+        [string]
+        $TenantName,
+        
+        [Parameter(Mandatory = $True)]
+        [string]
+        $TenantId,
+        
+        [string]
+        $ResourceUrl = 'https://analysis.windows.net/powerbi/api',
+        
+        [string]
+        $AuthUrl = 'https://login.windows.net/common/oauth2/authorize',
+        
+        [string]
+        $TokenUrl = 'https://login.microsoftonline.com/common/oauth2/token',
+        
+        [Parameter(Mandatory = $True)]
+        [string]
+        $RedirectUrls,
+        
+        [Alias('SqlServerInstance')]
+        [string]
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
+        [string]
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential
+	)
+	
+	if ($PSCmdlet.ShouldProcess($ReportServerInstance, "Registering PowerBI for SQL Server Instance"))
```

Comment:
Based on what you have in Backup-RSEncryptionKey, don't you need to add the computer name, instance name and the additional logic to make use of the Connect-RsReportServer (if it gets refactored you could just add the call here)